### ISSUE TITLE
Training from scratch ETA?

### ISSUE THREAD

**Issue by Unfixab1e** [ekzhang/fastseg#3]
Just wondering if you are still planning to give details about training fastseg from scratch?
Would be great!

**Comment by ekzhang** [comment]
Hi @Unfixab1e, this is currently blocking on updates to https://github.com/NVIDIA/semantic-segmentation to become public, and then I can release my training code, which is based on that repository. I'm not sure what the ETA on this is, but ADLR is currently working on it.

Happy to answer any questions about specific details though. I trained for 175 epochs with with [RMI Loss](https://github.com/ZJULearning/RMI), used standard data augmentation image transforms (random scaling and crop, horizontal flip, color jitter, Gaussian blur), used a linear learning rate schedule, started from [pretrained weights](https://github.com/rwightman/pytorch-image-models/) for the MobileNetV3 trunk, and used [uniform sampling](https://github.com/NVIDIA/semantic-segmentation/blob/4fb7b9e336816b5907bc615619b7bcab9eef1370/datasets/uniform.py) during the first 150 epochs of training.

Thanks!

**Comment by greeneggsandyaml** [comment]
Given the recent updates to the NVIDIA repo, do you think it may be possible to release the training code at this time? 

Thank you so much for this great work!

**Comment by ekzhang** [comment]
Thanks for the reminder @greeneggsandyaml! I didn't realize that they released the code, and also just one day after I made that last comment in the thread. 😄 

I'll start working on it today.

**Comment by greeneggsandyaml** [comment]
Amazing, thank you!

**Comment by Unfixab1e** [comment]
Great news, looking forward to try train myself~

**Comment by ekzhang** [comment]
I ported the training code over! You can try training it at https://github.com/ekzhang/semantic-segmentation - be warned though, that I haven't properly tested this, so there may be minor errors.

**Comment by ekzhang** [comment]
Hope this is useful to you as a reference :)

**Comment by greeneggsandyaml** [comment]
Amazing, thank you so much! This is great :)

**Comment by Unfixab1e** [comment]
Thanks a lot @ekzhang , it certainly is a great reference to me :)
I trained a deeplabv3+/mobilenetv3 (from tf model zoo) network on my custom dataset so I'm curious to compare it to fastseg and its pytorch implementation!
